### ISSUE TITLE
Dev suggestion: Allow DW-based sites to disable anonymous (logged-out) support requests

### ISSUE THREAD

**Issue by Sophira** [dreamwidth/dw-free#1630]
Apologies for using GitHub for a suggestion, but it doesn't really seem appropriate for **dw_suggestions** (as it's not something users would benefit from) and **dw_dev** seems a little bit too open for what could be a potentially sensitive request.

I believe that DW-based sites should be able to restrict support requests to logged-in users only if they choose. My reasoning is that I noticed that today the 'Daily Snapshot' Dreamhack was being bombarded with spammy support requests. Normally, I wouldn't care about this because the Daily Snapshot resets each day anyway.

In this case, however, there's a very important reason to be concerned - the support board automatically fires off a confirmation email when a support request is opened, and very often the email addresses used by spammers for the From and Return-Path email headers are real, working addresses of people who had nothing to do with it and who are thus very likely to mark the email as spam (because, technically speaking, it kind of <em>is</em>).

I could set `$LJ::CAPTCHA_FOR{support_submit_anon}` to 1 in `config.pl`, but that only works with an API key - and that's not something which I think would be wise to deploy over every Dreamhack. Apparently there's a demo key available, but the sort of usage I was seeing before I shut the Dreamhack down (571 support requests in 3 hours) would most likely be far beyond what would be allowed for a demo key.

Basically, I'd like to be able to have the default Dreamhack configuration disallow these anonymous support requests by default, as they end up sending spam unwittingly and I'm not okay with that.

I may come up with a pull request for this myself if people agree that this is worth doing, but I didn't want to solve a problem if people don't think it needs solving!

**Comment by kareila** [comment]
This seems to me to be in line with the other spam prevention measures documented at http://wiki.dreamwidth.net/wiki/index.php/Dreamhack_spam_prevention - seems like a reasonable default for dev servers in general, as long as it can be temporarily overridden for testing purposes.

**Comment by rahaeli** [comment]
Sure, I don't see why not. Maybe even a $IS_DEV_SERVER check that turns off anon support requests, period -- I don't see any reason why a dev server needs anon support requests enabled enough to make it a config variable.

**Comment by Sophira** [comment]
I think it's better as a separate variable - production sites might want to disable it too. Devs may also need to change this codepath, but I can see an argument for doing things that way.

I'll work on it. Claimed!'
